### ISSUE TITLE
Numeric input min prop is causing rounded value

### ISSUE THREAD

**Issue by tgreen7** [palantir/blueprint#4024]
#### Environment

- __Package version(s)__:  3.24.0
- __Browser and OS versions__: chrome Version 80.0.3987.149

If possible, link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/blueprint-sandbox-et9xy)): https://codesandbox.io/s/mutable-frog-kw5m3

#### Actual behavior

9.95 is rounded to 10 when min is set

#### Expected behavior

no rounding should occur because 9.95 > 0

**Comment by adidahiya** [comment]
What happens when you click the buttons though? And is it what you expected?

When you set the `min` prop, it looks like there is some extra processing that happens to round the value to the `minorStepSize` on the first render. But that same rounding would happen if a user clicked the increment/decrement buttons:

![2020-03-24 17 21 37](https://user-images.githubusercontent.com/723999/77478185-f4d06500-6df3-11ea-84b9-3d4a31b35d4d.gif)

It seems like you should be adjusting your step size if you want to supply a value like 9.95. If you expect some other behavior, let me know...

**Comment by adidahiya** [comment]
Looking a little more closely... I can see it's a little confusing how we derive the value precision from the step size...

https://github.com/palantir/blueprint/blob/803cf2c036c0f58cf5696b571cd21ce769facc72/packages/core/src/components/forms/numericInput.tsx#L237

but do you have any proposals for an alternative? for example, in your original code example, do you expect users to be selecting values as follows: 9.95, 10.95, 11.95, .... ?

**Comment by tgreen7** [comment]
@adidahiya I wouldn't want there to be any rounding initially. Basically I have a form which is getting initialized from data and I would want the user to see the exact values coming in. I am not sure why this component would do any rounding on that initial render. Is there some way to get around that?

**Comment by adidahiya** [comment]
> I would want the user to see the exact values coming in.

Yes, I get that, and the current behavior is surprising from a developer perspective... but still think there's potential for confusion after the initial render in your proposed use case because rounding will occur after the user clicks increment/decrement.

You can work around this by setting the step size props, like `minorStepSize={0.05}`
